src/dotless.Core/Parser/Tree/Ruleset.cs

```diff
@@ -279,6 +281,50 @@ protected void EvaluateRules(Env env)
             env.Frames.Pop();
         }
 
+        /// <summary>
+        /// Merges properties with merge operators '+' and '+_'
+        /// </summary>
+        /// <returns>New merged ruleset</returns>
+        private NodeList MergeProperties()
+        {
+            var groups = new Dictionary<string, List<Rule>>();
+            var sorted = new List<Node>();
+            foreach (var node in Rules)
+            {
+                var rule = node as Rule;
+                if (rule == null || rule.Merge == "")
```

Comment:
Performance implications aside, you could pull out the rules that match this condition by doing a

```csharp
var rulesToMerge = Rules.OfType<Rule>().Where(r => !string.IsNullOrEmpty(r.Merge));
var sorted = Rules.Except(rulesToMerge).Cast<Node>().ToList()
```

prior to the loop.